packages/console/src/components/CreateTenantModal/SelectTenantPlanModal/PlanCardItem/index.tsx

```diff
@@ -0,0 +1,120 @@
+const featuredQuotaKeys: Array<keyof SubscriptionPlanQuota> = [
+  'mauLimit',
+  'machineToMachineLimit',
+  'standardConnectorsLimit',
+  'rolesLimit',
+  'scopesPerRoleLimit',
+  'auditLogsRetentionDays',
+];
+
+type Props = {
+  plan: SubscriptionPlan;
+  onSelect: () => Promise<void>;
+};
+
+function PlanCardItem({ plan, onSelect }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console.upsell.create_tenant' });
+  const { tenants } = useContext(TenantsContext);
+  const { stripeProducts, id: planId, name: planName, quota } = plan;
+
+  const basePrice =
+    stripeProducts.find(({ type }) => type === 'flat')?.price.unitAmountDecimal ?? 0;
+
+  const tierPrices = stripeProducts
+    .filter(({ type }) => type !== 'flat')
+    .map(({ price }) => {
+      return price.unitAmountDecimal;
+    });
+
+  const isFreePlan = planId === ReservedPlanId.free;
+
+  // Todo: @xiaoyijun filter our all free tenants
+  const isFreeTenantExceeded = tenants.length >= maxFreeTenantLimit;
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.planInfo}>
+        <div className={styles.title}>
+          <PlanName name={planName} />
+        </div>
+        <div className={styles.priceInfo}>
+          <div className={styles.priceLabel}>{t('base_price')}</div>
+          <div className={styles.price}>
+            ${t('monthly_price', { value: Number(basePrice) / 100 })}
+          </div>
+          <div className={styles.priceLabel}>
+            {t('mau_unit_price')}
+            <span className={styles.unitPrices}>
+              {tierPrices.length > 0
+                ? tierPrices.map((price) => `$${Number(price) / 100}`).join(' ')
+                : '$0.00'}
+            </span>
+          </div>
+        </div>
+        <div className={styles.description}>
+          <PlanDescription planName={planName} />
+        </div>
+      </div>
+      <div className={styles.content}>
+        <PlanQuotaList
+          hasIcon
+          quota={quota}
+          featuredQuotaKeys={featuredQuotaKeys}
+          className={styles.list}
+        />
+        {isFreePlan && isFreeTenantExceeded && (
+          <div className={classNames(styles.tip, styles.exceedFreeTenantsTip)}>
+            {t('free_tenants_limit', { count: maxFreeTenantLimit })}
+          </div>
+        )}
+        {!isFreePlan && (
+          <div className={styles.tip}>
+            <TextLink
+              isTrailingIcon
+              href="https://logto.io/pricing"
+              icon={<ArrowRight className={styles.linkIcon} />}
+              target="_blank"
+              className={styles.link}
+            >
+              <DynamicT forKey="upsell.create_tenant.view_all_features" />
+            </TextLink>
+          </div>
+        )}
+        <Button
+          title={
+            <DangerousRaw>
+              <Trans components={{ name: <PlanName name={planName} /> }}>
+                {t(isFreePlan ? 'select_plan' : 'upgrade_to')}
```

Comment:
Is this always true? If the user is currently on "Pro" plan and then the "Hobby" plan should be "downgrade to" instead of "upgrade to", right?

Reply:
No, this is for a new tenant, not current tenant.